### ISSUE TITLE
Windows 11 notifications missing in action center

### ISSUE THREAD

**Issue by Toastbroti** [Betterbird/thunderbird-patches#273]
Using native notifications on Windows 11 they will show up in the corner and then disappear, without being added to the notifications in the action center.
![grafik](https://github.com/Betterbird/thunderbird-patches/assets/39853282/ff875655-b17d-4394-83d1-bf781a22b4f2)
Thunderbird doesn't have this problem.

**Comment by Betterbird** [comment]
Sigh. Notifications can't be clicked on in TB, see
https://bugzilla.mozilla.org/show_bug.cgi?id=1863798
https://bugzilla.mozilla.org/show_bug.cgi?id=1858603
https://bugzilla.mozilla.org/show_bug.cgi?id=1840123

We fixed that issue by rolling back the entire notifications code to the state of 102 when it still worked.

And now you're saying the notifications don't get added to the "action centre" in Win 11. Well, last we checked, they got added to the Win 10 "action centre", but sending some test message just now, they didn't get added. So it's quite confusing.

**Comment by Betterbird** [comment]
@Toastbroti ?

**Comment by Toastbroti** [comment]
![grafik](https://github.com/Betterbird/thunderbird-patches/assets/39853282/9ec49ef7-a4ae-4acb-8dd7-24a55bee402f)
It's both turned on but messages aren't added to the action center.

**Comment by Betterbird** [comment]
I had an exchange with a user who compared TB to BB on Windows 11. He told me that the notifications end up in the Action Center but get removed after a few seconds. Apparently it's related to whether you click on the notification bell at the bottom left:
![image](https://github.com/Betterbird/thunderbird-patches/assets/86832723/c42e490a-26a8-4f68-ac2b-9b61d9c8eb6a)

In any case, TB and BB are different in this regard since the BB we fixed the "clicking on banner doesn't activate the message" issue. We'll how all this pans out in the next version.

**Comment by Toastbroti** [comment]
Notifications get added to the center only if you open it before or while the banner is shown, and as soon as you close the center they disappear.
Seems like all we can do is wait for TB to fix the bug that got me to switch to BB in the first place.

**Comment by Betterbird** [comment]
Which bug is that? The activation bug? It works fine in Windows 10: Activation, and the notifications end up in the action center.

IMHO, the activation is more important than the stuff accumulating in the action center.

**Comment by Betterbird** [comment]
Please try this build:
https://www.betterbird.eu/downloads/WindowsInstaller/betterbird-115.9.0-bb26-latest-build2.en-US.win64.installer.exe

Notifications should show in the action centre and you should even be able to activate BB. As a trade off, clicking a banner notification will no longer bring a non-minimised window to the front.

**Comment by Betterbird** [comment]
Update:
Try
https://www.betterbird.eu/downloads/WindowsInstaller/betterbird-115.9.0-bb26-latest-build3.en-US.win64.installer.exe
That should also fix bringing the window to the foreground.

**Comment by Toastbroti** [comment]
Works perfectly, thanks!

**Comment by Betterbird** [comment]
Thanks for testing, we'll ship it "officially" in the next release 115.9.1 or 115.10.0.

**Comment by Betterbird** [comment]
@Toastbroti : Are you aware that those notifications are actually called Toast?

https://learn.microsoft.com/en-us/windows/apps/design/shell/tiles-and-notifications/toast-notifications-overview

**Comment by Toastbroti** [comment]
Yes haha

**Comment by mehlian** [comment]
Clicking on notification when new mail appears to open this mail in Betterbird no longer works in latest version (115.13.0-bb30 (64-bit)). It worked in 115.6.0-bb21 (64-bit).
OS: Windows 10.

**Comment by Betterbird** [comment]
You're talking about the Windows system notification? Tested in 115.13.0-bb30 just now and working fine.

**Comment by mehlian** [comment]
> You're talking about the Windows system notification? Tested in 115.13.0-bb30 just now and working fine.

Yes, just tested it again. Notification appears, but clicking on it does nothing. Notification just disapears.

**Comment by mehlian** [comment]
I did not change that.
alerts.useSystemBackend = true
alerts.useSystemBackend.windows.notificationserver.enabled = false

By notification I mean small popup with short description above system clock.

**Comment by Betterbird** [comment]
> * Update Thunderbird to latest version by opening Thunderbird and clicking 'Help>About>Update' button
> 
> * Update Betterbird to latest version by downloading newest version from webpage and using installation wizzard update to latest version.

It's known to be broken in TB 115, they fixed it for TB 128.

It's true that we switched the Windows system notification to different code in 115.10.0, see release notes:
CHANGED - Switched Windows system notifications to new Mozilla platform code to improve registering in the Windows Action Centre on Windows 11 (fixed in 115.10.0)

That was done to fix the notifications for Windows 11, and the reporter of this ticket confirmed it working (https://github.com/Betterbird/thunderbird-patches/issues/273#issuecomment-2019513839).

Of course we tested that the new way would also work on Windows 10. Maybe you have access to a different machine to try it there. As I said, it works here and yours is the only report that it doesn't work on Windows 10.
